### ISSUE TITLE
CRDTs

### ISSUE THREAD

**Issue by staab** [mikedilger/nostr-next#34]
Replaceable events in general are bad, but especially lists, which can have race conditions when used from multiple clients. p2panda has an excellent design which accounts for this.

**Comment by mikedilger** [comment]
Can you summarize how p2panda solves this?

**Comment by staab** [comment]
[Entries](https://p2panda.org/learn/entries) are like nostr events, with a hash + signature, with the following differences:

- Payloads are separate from entries, so you can download histories without downloading content.
- They provide a `backlink` to the previous entry in the log.
- `log`s (merkle trees representing a series of entries) are differentiated by a `log id`, reducing contention over histories.
- Entries have sequence numbers, which can help with detecting and reconciling forks.
- Conflicts are resolved deterministically, resulting in eventual consistency. They can also be resolved manually with a [merge](https://p2panda.org/learn/operations#operation-graphs) operation.
